lib/elixir/lib/gen_event.ex

```diff
@@ -655,11 +655,12 @@ defmodule GenEvent do
   Record.defrecordp :handler, [:module, :id, :state, :pid, :ref]
 
   @doc false
-  def init_it(starter, :self, name, mod, args, options) do
+  def init_it(starter, :self, name, mod, args, options)
+      when is_atom(mod) and is_list(args) and is_list(options) do
```

Comment:
We set the `mod` and `args` so we don't need to check these. Also `options` must be a list otherwise `do_start` would crash. If we were to check it, we would want to check these on the boundary of the API, not in an internal callback.